cpp/src/interpolate/cubic_spline.cu

```diff
@@ -0,0 +1,405 @@
+    cudf::experimental::fill_in_place(Dlu_buffer, 0, Dlu_col->size(), zero);
+    cudf::experimental::fill_in_place(u_buffer, 0, u_col->size(), zero);
+    
+    TPRINT(h_buffer, "h_zero");
+    TPRINT(D_buffer, "D_one");
+    TPRINT(Dlu_buffer, "Dlu_zero");
+    cudf::experimental::type_dispatcher(y.type(), compute_spline_tridiagonals{}, t, y, prefixes, D_buffer, Dlu_buffer, u_buffer, h_buffer, i_buffer, mr, stream);
+    //compute_spline_tridiagonals comp_tri;
+    //comp_tri.operator()<float>(t, y, prefixes, D_buffer, Dlu_buffer, u_buffer, h_buffer, i_buffer, mr, stream);
+
+    TPRINT(h_buffer, "h_i");
+    TPRINT(i_buffer, "i_i");
+    TPRINT(D_buffer, "D_i");
+    TPRINT(Dlu_buffer, "Dlu_i");
+    TPRINT(u_buffer, "u_i");
+
+    // cusparse solve n length m tridiagonal systems
+    // 4. call cusparse<T>gtsv2() to solve
+    // 4.1 Get cuSparse library context
+    // compute inputs:
+    // handle: the cuSparse library context
+    // m: size
+    // n: number of columns of solution matrix B
+    // dl, d, du: vectors of the diagonal
+    // B: (ldb, n) dimensional dense matrix to be solved for
+    // ldb: leading dimension of B
+    // pBuffer: get size of thisu by gtsv2_bufferSizeExt 
+    cusparseStatus_t cusparseStatus;
+    cusparseHandle_t handle;
+    cudaMalloc(&handle, sizeof(cusparseHandle_t));
+    cusparseStatus = cusparseCreate(&handle);
+    detail::HANDLE_CUSPARSE_STATUS(cusparseStatus);
+    size_t pBufferSize;
+    cusparseStatus = cusparseSgtsv2StridedBatch_bufferSizeExt(
+        handle,
+        y.size()/(prefixes.size()-1),
+        Dll_buffer.data<float>(),
+        D_buffer.data<float>(),
+        Dlu_buffer.data<float>(),
+        u_buffer.data<float>(),
+        prefixes.size()-1,
+        y.size()/(prefixes.size()-1),
+        &pBufferSize
+    );
+    detail::HANDLE_CUSPARSE_STATUS(cusparseStatus);
+    rmm::device_vector<float> pBuffer(pBufferSize);
+    cusparseStatus = cusparseSgtsv2StridedBatch(
+        handle,
+        y.size()/(prefixes.size()-1),
+        Dll_buffer.data<float>(),
+        D_buffer.data<float>(),
+        Dlu_buffer.data<float>(),
+        u_buffer.data<float>(),
+        prefixes.size()-1,
+        y.size()/(prefixes.size()-1),
+        thrust::raw_pointer_cast(pBuffer.data())
+    );
+    detail::HANDLE_CUSPARSE_STATUS(cusparseStatus);
+    cusparseStatus = cusparseDestroy(handle);
+    detail::HANDLE_CUSPARSE_STATUS(cusparseStatus);
+
+    int dn = n - (prefixes.size()-1);
+    // Finally, compute coefficients via Horner's scheme
+    auto d3_col = make_numeric_column(y.type(), dn, cudf::mask_state::UNALLOCATED, stream, mr);
+    auto d2_col = make_numeric_column(y.type(), dn, cudf::mask_state::UNALLOCATED, stream, mr);
+    auto d1_col = make_numeric_column(y.type(), dn, cudf::mask_state::UNALLOCATED, stream, mr);
+    auto d0_col = make_numeric_column(y.type(), dn, cudf::mask_state::UNALLOCATED, stream, mr);
+    cudf::mutable_column_view d3 = d3_col->mutable_view();
```

Comment:
prefer `auto` in simple cases to avoid unintentional implicit type casting.

`auto d3 = d3_col->mutable_view();`